includes/abstracts/llms-abstract-posts-query.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * LLMS_Abstract_Posts_Query class file
+ *
+ * @package LifterLMS/Abstracts/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Abstract WP_Posts query class.
+ *
+ * This class is meant to perform custom queries that ultimately
+ * get passed into a `WP_Query`, ideally for a specific post type
+ * or list of post types.
+ *
+ * @since [version]
+ *
+ * Valid query arguments
+ *
+ * {@see LLMS_Abstract_Query} for inherited query arguments.
+ *
+ * @param string          $fields     WP_Post fields to return for each result. Accepts "all", "ids", or "id=>parent". Default: "all".
+ * @param string[]|string $status     Limit results by WP_Post `$post_status`. Default: "publish".
+ * @param string[]        $post_types Limit results to the specified post type(s).
+ */
+abstract class LLMS_Abstract_Posts_Query extends LLMS_Abstract_Query {
+
+	/**
+	 * Defines fields that can be sorted on via ORDER BY.
+	 *
+	 * @var string[]
+	 */
+	protected $allowed_sort_fields = array(
+		'ID',
+		'author',
+		'title',
+		'name',
+		'type',
+		'date',
+		'modified',
+		'parent',
+		'menu_order',
+	);
+
+	/**
+	 * Specify the post types allowed to be queried by this class
+	 *
+	 * This array should be a list of one or more post type names.
+	 *
+	 * @var string[]
+	 */
+	protected $allowed_post_types = array();
+
+	/**
+	 * The WP_Query instance.
+	 *
+	 * @var null
+	 */
+	protected $wp_query = null;
+
+	/**
+	 * Set result counts and pagination properties.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	protected function count_results() {
+
+		$this->number_results = $this->wp_query->post_count;
+		$this->found_results  = $this->found_results();
+		$this->max_pages      = $this->wp_query->max_num_pages;
```

Comment:
```suggestion
		$this->max_pages      = (int) $this->wp_query->max_num_pages;
```
Since `$this->wp_query->max_num_pages` seems to be a `float`, let's cast this (for future comparisons, e.g. in `is_last_page()` method).